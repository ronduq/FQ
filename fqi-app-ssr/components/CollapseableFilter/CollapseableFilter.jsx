import React from 'react';
import classNames from 'classnames';

import Checkbox from '../Checkbox/Checkbox';
import Icon from '../Icon/Icon';
import s from './CollapseableFilter.module.scss';
import t from '../Icon/Icon.module.scss';

class CollapseableFilter extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      open: true,
      allSelected: true
    }
    this.togglePanel = this.togglePanel.bind(this);
    this.selectAll = React.createRef();
  }

  togglePanel(e){
    this.setState({open: !this.state.open})
  }

  listItems() {
    const { 
      data, 
      updateItem,
      currentRetailer,
      criteria
    } = this.props;
    const renderedList = Object.keys(data.items).map((item, index) => {
      const element = data.items[item];
      const labelItem = data.itemsLabels[element.id];
      const label = labelItem == null ? element.id : labelItem;
      let type = '';

      if (element.isChild) {
        type = 'child';
      } else if (element.isParent) {
        type = 'parent';
      }

      return (
        <li className={classNames(s.listItem, {[s.listItemMain]: element.isParent} )} key={index}>
          <Checkbox 
            label={label} 
            type={type}
            updateItem={updateItem}
            index={index}
            selected={element.selected}
            parentCode={element.parentCode}
            isParent={element.isParent}
            disabled={currentRetailer && currentRetailer === element.retailerCode}
            currentRetailer={currentRetailer}
            criteria={criteria}
          />
        </li>
      )
    });
    return renderedList;
  }

  render() {
    const { data, selectAllItems, title } = this.props;

    return (
      <fieldset>
        <button onClick={(e)=>this.togglePanel(e)} className={s.header} aria-expanded={this.state.open}>
          <Icon className={classNames(s.titleIcon, t.iconMedium, t.iconTeal)} iconName={data.name} />
          <legend className={s.title}>{title}</legend>
          <Icon className={classNames(s.arrowIcon, t.iconSmall, t.iconOrange, {[s.arrowDown]: this.state.open})} iconName={`chevron`} />
        </button>
        {this.state.open ? (
          <ul className={s.list}>
            <li className={classNames(s.listItem, s.listItemMain)}>
              <Checkbox 
                ref={this.selectAll} 
                label='All' 
                selected={data.allItemsSelected} 
                type='all'
                allNumber = {data.count || Object.keys(data.items).length}
                selectAllItems={selectAllItems}
                name={data.name}
              />
            </li>
            {this.listItems()}
          </ul>
        ) : null}
      </fieldset>
    );
  }
}

export default CollapseableFilter;
  
  