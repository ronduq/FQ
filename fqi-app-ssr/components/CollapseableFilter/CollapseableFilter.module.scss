@import '../../style/settings/colours';
@import '../../style/settings/fonts';

.header{
  cursor: pointer;
  background: white;
  border:none;
  border-bottom: solid 1px $colour-grey-medium;
  padding: 15px;
  height: 60px;
  width:100%;
  @include font-family('proxima-light');
  @include font-size('medium');

  &:focus {
    outline: solid #32a1ce;
  }

  &:hover {
    .title {
      text-decoration: underline;
    }
  }
}

.content{
  cursor: pointer;
  padding: 15px;
}

.title {
  padding-left: 10px;
  float: left;
  transform: translateY(2px);
}

.titleIcon {
  transform: translateY(2px);
  float: left;
}

.arrowIcon {
  float: right;
  transform: translateY(4px) rotate(180deg);
  transition: ease-in-out transform .2s;
}

.arrowDown {
  transform: translateY(4px);
}

.list {
  list-style-type: none;
  padding: 15px;
  animation: fadein .4s;
}

.listItem {
  display: inline-block;
  margin-bottom: 10px;
  width: 50%;
  vertical-align: top;
}

.listItemMain {
  display: block;
}

// Animation
@keyframes fadein {
  from { 
    opacity: 0; 
    transform: translateY(-30px);
  }
  to   { 
    opacity: 1; 
    transform: translateY(0);
  }
}