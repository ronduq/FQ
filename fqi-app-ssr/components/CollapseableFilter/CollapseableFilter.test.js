/* eslint-env jest */

import { mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import CollapseableFilter from './CollapseableFilter'

const data = {
  allItemsSelected: true,
  items: {
    '0': {
      id: "avocado",
      isChild: true,
      isParent: false,
      produceCode: "avocado",
      produceName: "Avocado",
      selected: true,
    },
    '1': {
      id: "banana",
      isChild: false,
      isParent: true,
      produceCode: "banana",
      produceName: "Banana",
      selected: true,
    },
  },
  itemsImages: {
    avocado: "Avocado",
    banana: 'Banana',
  },
  itemsLabels: {
    avocado: "Avocado",
    banana: 'Banana',
  }
}

const togglePanel = jest.fn();

describe('With Snapshot Testing', () => {
  it('CollapseableFilter snapshot"', () => {
    const component = renderer.create(<CollapseableFilter data={data}/>)
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  });

  it('should call updateItem when button clicked', () => {
    const wrapper = mount(<CollapseableFilter  data={data}/>);
    const button = wrapper.find('button');
    button.simulate('click');
    expect(togglePanel).toHaveBeenCalledTimes(0);
  });
})