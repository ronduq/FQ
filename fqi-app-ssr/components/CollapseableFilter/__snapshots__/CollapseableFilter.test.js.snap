// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing CollapseableFilter snapshot" 1`] = `
<fieldset>
  <button
    aria-expanded={true}
    className="header"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="titleIcon iconMedium iconTeal"
      focusable="false"
    >
      <use
        xlinkHref="#undefined"
      />
    </svg>
    <legend
      className="title"
    />
    <svg
      aria-hidden="true"
      className="arrowIcon iconSmall iconOrange arrowDown"
      focusable="false"
    >
      <use
        xlinkHref="#chevron"
      />
    </svg>
  </button>
  <ul
    className="list"
  >
    <li
      className="listItem listItemMain"
    >
      <div
        className="checkbox"
      >
        <input
          autoComplete="off"
          checked={true}
          id="Allundefined"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          aria-hidden="true"
          htmlFor="Allundefined"
        >
          <span>
            All
             (2)
          </span>
        </label>
      </div>
    </li>
    <li
      className="listItem"
    >
      <div
        className="checkbox checkboxChild"
      >
        <input
          autoComplete="off"
          checked={true}
          id="Avocadoundefined"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          aria-hidden="true"
          htmlFor="Avocadoundefined"
        >
          <span>
            Avocado
            
          </span>
        </label>
      </div>
    </li>
    <li
      className="listItem listItemMain"
    >
      <div
        className="checkbox checkboxParent"
      >
        <input
          autoComplete="off"
          checked={true}
          id="Bananaundefined"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          aria-hidden="true"
          htmlFor="Bananaundefined"
        >
          <span>
            Banana
            
          </span>
        </label>
      </div>
    </li>
  </ul>
</fieldset>
`;
